pkg/controllers/state/cluster.go

```diff
@@ -55,7 +55,8 @@ type Cluster struct {
 	kubeClient    client.Client
 	cloudProvider cloudprovider.CloudProvider
 	clock         clock.Clock
-	hasSynced     atomic.Bool
+	// We use this value to determine if we have synced the cluster state atleast once
+	HasSynced atomic.Bool
```

Comment:
I'd expose this through a `Getter` rather than the raw atomic - we don't want any other part of the codebase updating this.